Core/include/Acts/EventData/detail/residual_calculator.hpp

```diff
@@ -24,43 +25,59 @@ namespace detail {
 ///
 /// @return residual_calculator<params...>::result(first,second) yields the
 ///         residuals of `first` with respect to `second`
-template <ParID_t... params>
+template <typename parameter_indices_t, unsigned int... params>
```

Comment:
Why does the parameter pack have to be `unsigned int` here?

Reply:
The `ParID_t` is the old `BoundParametersIndices` so initially I copied the code here and did the same for `FreeParametersIndices`. However both are convertible to `unsigned int`, so I pass the `unsigned int` around to make the code reusable for different parameter indices.

Reply:
Oh wait, let me check if isn't possible to change it

Reply:
Right, figured it out and changed `unsigned int`--> `parameter_indices_t` (includes less casts)